src/legacy/core_plugins/data/public/index.ts

```diff
@@ -38,12 +38,16 @@ class DataService {
 }
 
 /**
- * We temporarily export default here so that users importing from 'plugins/data'
+ * We export data here so that users importing from 'plugins/data'
  * will automatically receive the response value of the `setup` contract, mimicking
  * the data that will eventually be injected by the new platform.
  */
-// eslint-disable-next-line import/no-default-export
-export default new DataService().setup();
+export const data = new DataPlugin().setup();
 
 /** @public */
-export type DataSetup = ReturnType<DataService['setup']>;
+export interface DataSetup {
+  indexPatterns: IndexPatternsSetup;
+}
+
+/** @public types */
+export { IndexPattern, StaticIndexPattern, StaticIndexPatternField, Field } from './index_patterns';
```

Comment:
Is this export pattern to be use only here or for all the sub-components as well?